openmeter/customer/event.go

```diff
@@ -0,0 +1,145 @@
+			New:          &customer,
+		},
+	}
+}
+
+// CustomerCreateEvent is an event that is emitted when a customer is created
+type CustomerCreateEvent struct {
+	metadata.EventEntityMutationPayload[Customer] `json:",inline"`
+}
+
+func (e CustomerCreateEvent) EventName() string {
+	return metadata.GetEventName(metadata.EventType{
+		Subsystem: CustomerEventSubsystem,
+		Name:      CustomerCreateEventName,
+		Version:   "v1",
+	})
+}
+
+func (e CustomerCreateEvent) EventMetadata() metadata.EventMetadata {
+	return metadata.EventMetadata{
+		ID:     e.New.ID,
+		Source: metadata.ComposeResourcePath(e.New.Namespace, metadata.EntityCustomer, e.New.ID),
+		Time:   e.New.CreatedAt,
+	}
+}
+
+func (e CustomerCreateEvent) Validate() error {
+	var errs []error
+
+	if err := e.EventEntityMutationPayload.Validate(); err != nil {
+		errs = append(errs, fmt.Errorf("customer create event: %w", err))
+	}
+
+	return errors.Join(errs...)
+}
+
+// NewCustomerUpdateEvent creates a new customer update event
+func NewCustomerUpdateEvent(new Customer, previous Customer) CustomerUpdateEvent {
+	return CustomerUpdateEvent{
+		EventEntityMutationPayload: metadata.EventEntityMutationPayload[Customer]{
+			Entity:       metadata.EntityCustomer,
+			MutationType: metadata.EntityMutationTypeUpdate,
+			New:          &new,
+			Previous:     &previous,
+		},
+	}
+}
+
+// CustomerUpdateEvent is an event that is emitted when a customer is updated
+type CustomerUpdateEvent struct {
+	metadata.EventEntityMutationPayload[Customer] `json:",inline"`
+}
+
+func (e CustomerUpdateEvent) EventName() string {
+	return metadata.GetEventName(metadata.EventType{
+		Subsystem: CustomerEventSubsystem,
+		Name:      CustomerUpdateEventName,
+		Version:   "v1",
+	})
+}
+
+func (e CustomerUpdateEvent) EventMetadata() metadata.EventMetadata {
+	return metadata.EventMetadata{
+		ID:     metadata.GetMutationEventID(metadata.EntityMutationTypeUpdate, e.Previous.ID, e.New.UpdatedAt),
+		Source: metadata.ComposeResourcePath(e.New.Namespace, metadata.EntityCustomer, e.New.ID),
+		Time:   e.New.UpdatedAt,
+	}
+}
+
+func (e CustomerUpdateEvent) Validate() error {
+	var errs []error
+
+	if err := e.EventEntityMutationPayload.Validate(); err != nil {
+		errs = append(errs, fmt.Errorf("customer update event: %w", err))
+	}
+
+	return errors.Join(errs...)
+}
+
+// NewCustomerDeleteEvent creates a new customer delete event
+func NewCustomerDeleteEvent(previous Customer, deletedAt time.Time) CustomerDeleteEvent {
+	return CustomerDeleteEvent{
+		EventEntityMutationPayload: metadata.EventEntityMutationPayload[Customer]{
+			Entity:       metadata.EntityCustomer,
+			MutationType: metadata.EntityMutationTypeDelete,
+			Previous:     &previous,
+		},
+		deletedAt: deletedAt,
+	}
+}
+
+// CustomerDeleteEvent is an event that is emitted when a customer is deleted
+type CustomerDeleteEvent struct {
+	metadata.EventEntityMutationPayload[Customer] `json:",inline"`
+	deletedAt                                     time.Time
```

Comment:
```suggestion
	Customer `json:",inline"`
```

Customer has a deleted at field.